src/initialize.F90

```diff
@@ -902,4 +916,230 @@ subroutine allocate_banks()
 
+!===============================================================================
+! PREPARE_DISTRIBCELL initializes any distribcell filters present and sets the
+! offsets for distribcells
+!===============================================================================
+
+  subroutine prepare_distribcell()
+
+    integer :: i, j       ! Tally, filter loop counters
+    integer :: extra      ! Number of extra filters to add
+    integer :: n_filt     ! Number of filters originally in tally
+    logical :: count_all  ! Count all cells
+    type(TallyObject),    pointer :: tally            ! Current tally
+    type(Universe),       pointer :: univ             ! Pointer to universe
+    type(Cell),           pointer :: c                ! Pointer to cell
+    integer, allocatable :: univ_list(:)              ! Target offsets
+    integer, allocatable :: kounts(:,:)               ! Target count
+    logical, allocatable :: found(:,:)                ! Target found
+
+    count_all = .false.
+
+    ! Loop over tallies    
+    do i = 1, n_tallies
+
+      extra = 0
+
+      ! Get pointer to tally
+      tally => tallies(i)
+
+      n_filt = tally % n_filters
+
+      ! Loop over the filters to determine how many additional filters
+      ! need to be added to this tally
+      do j = 1, tally % n_filters
+
+        ! Determine type of filter
+        if (tally % filters(j) % type == FILTER_DISTRIBCELL) then
+          count_all = .true.
+          extra = extra + size(tally % filters(j) % int_bins) - 1
+        end if
+
+      end do
+
+      if (extra > 0) then
+        call fatal_error("At least one Distribcell filter was specified with &
+                         &multiple bins. This feature is not yet supported.")
+      end if      
+
+    end do
+    
+    if (count_all) then
+      
+      univ => universes(BASE_UNIVERSE)
+
+      ! sum the number of occurrences of all cells
+      call count_instance(univ)
+
+      ! Loop over tallies    
+      do i = 1, n_tallies    
+
+        ! Get pointer to tally
+        tally => tallies(i)      
+
+        ! Initialize the filters
+        do j = 1, tally % n_filters
+
+          ! Set the number of bins to the number of instances of the cell
+          if (tally % filters(j) % type == FILTER_DISTRIBCELL) then
+            c => cells(tally % filters(j) % int_bins(1))
+            tally % filters(j) % n_bins = c % instances
+          end if
+
+        end do
+      end do
+
+    end if
+
+    ! Allocate offset maps at each level in the geometry
+    call allocate_offsets(univ_list, kounts, found)
+
+    ! Calculate offsets for each target distribcell
+    do i = 1, n_maps
+      do j = 1, n_universes  
+        univ => universes(j)
+        call calc_offsets(univ_list(i), i, univ, kounts, found)
+      end do
+    end do
+
+    ! Deallocate temporary target variable arrays
+    deallocate(kounts)
+    deallocate(found)
+    deallocate(univ_list)
+  
+  end subroutine prepare_distribcell
+
+
+  
```

Comment:
One empty line separating a subroutine and the next comment block is the prevailing style in OpenMC code.